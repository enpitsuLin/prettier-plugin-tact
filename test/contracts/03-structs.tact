import "@stdlib/deploy";

struct Point {
    x: Int as int64;
    y: Int as int64;
}

struct Params {
    name: String = "Satoshi";   // default value
    age: Int? = null;           // optional field
    point: Point;               // nested structs
}

message Add {
    point: Point;               // message can hold a struct
}

contract Structs with Deployable {

    // contract persistent state variables
    s1: Point;
    s2: Params;

    init() {
        self.s1 = Point{x: 2, y: 3};
        self.s2 = Params{point: self.s1};
    }

    receive("show ops") {
        // temporary variable
        let s: Point = Point{x: 4, y: 5};

        self.s1 = s;
    }

    receive(msg: Add) {
        self.s1.x = self.s1.x + msg.point.x;
        self.s1.y = self.s1.y + msg.point.y;
    }

    get fun point(): Point {
        return self.s1;
    }

    get fun params(): Params {
        return self.s2;
    }
}