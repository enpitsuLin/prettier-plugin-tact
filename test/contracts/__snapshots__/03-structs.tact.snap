import "@stdlib/deploy";

struct Point {
    x: Int as int64;
    y: Int as int64;
}
struct Params {
    name: String = "Satoshi"; // default value
    age: Int? = ; // optional field
    point: Point; // nested structs
}

message Add {
    point: Point; // message can hold a struct
}

contract Structs with Deployable {
    // contract persistent state variables
    s1: Point;s2: Params;

    receive("show ops") {
        // temporary variable
        let s: Point = ;

        self.s1 = s;
    }

    receive(msg: Add) {
        self.s1.x = self.s1.x + msg.point.x;
        self.s1.y = self.s1.y + msg.point.y;
    }

    get fun point(): Point {
        return self.s1;
    }

    get fun params(): Params {
        return self.s2;
    }
}